x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,718 @@
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. Using your existing CA, generate a keystore for your nodes. You must
+use the CA that was used to sign the certificate currently in use.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `--ca <ca_file>`:: Name of the CA keystore used to sign your certificates.
+   If you used the `elasticsearch-certutil` tool to generate your existing CA,
+   the keystore name defaults to `elastic-stack-ca.p12`.
+====
+
+   a. Enter a name for the output file or accept the default of
+   `elastic-certificates.p12`.
+
+   b. When prompted, enter a password for the node keystore.
+
+. If you entered a password when creating the node keystore that is different 
+from the current keystore password, run the following command to store the
+password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+--
+
+. [[start-rolling-restart,step 4]]On the current node in your cluster where you're updating the keystore,
+start a <<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. [[replace-keystores]]Replace your existing keystore with the new keystore,
+ensuring that the file names match. For example, `elastic-certificates.p12`.
++
+IMPORTANT: If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. Start the node where you updated the keystore.
+
+. [[verify-keystore,step 8]](Optional) Use the
+<<security-api-ssl,SSL certificate API>> to verify that {es} loaded the new
+keystore.
+
+. If you're only updating certificates for the transport layer (and not the HTTP layer), then complete <<start-rolling-restart>> through <<verify-keystore>> one node at a time until you've updated all keystores in your cluster. You can then
+complete the remaining steps for a <<restart-cluster-rolling,rolling restart>>.
++
+Otherwise, do not complete a rolling restart and proceed to the steps for
+generating a new certificate for the HTTP layer.
```

Comment:
This reads a little strangely to me.

```suggestion
Otherwise, do not complete a rolling restart. Instead, proceed to the steps for
generating a new certificate for the HTTP layer.
```

Reply:
+1 to being more explicit. We're using this statement in two places (one for same CA and one for new CA), so I'll make your suggested change in both instances.